### ISSUE TITLE
[swarm] load-balancing not working as expected

### ISSUE THREAD

**Issue by rogaha** [docker/docker#25394]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:04:48 2016
 OS/Arch:      darwin/amd64
 Experimental: true

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 23:54:00 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 2
 Running: 2
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 11
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host null bridge overlay
Swarm: active
 NodeID: 2jml8zh2ap8gnw3ghchc03g09
 Is Manager: true
 ClusterID: 1h8ug9o7h7ihgspy7wmgr7z68
 Managers: 2
 Nodes: 5
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot interval: 10000
  Heartbeat tick: 1
  Election tick: 3
 Dispatcher:
  Heartbeat period: 5 seconds
 CA configuration:
  Expiry duration: 3 months
 Node Address: 192.168.99.100
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.16-boot2docker
Operating System: Boot2Docker 1.12.0 (TCL 7.2); HEAD : e030bab - Fri Jul 29 00:29:14 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 995.9 MiB
Name: master1
ID: XFBC:QIPK:ZJH5:MGSZ:D6IA:32XG:TUHL:6E43:HXOQ:FVLW:OY64:HWD4
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 55
 Goroutines: 212
 System Time: 2016-08-04T08:18:26.116262192Z
 EventsListeners: 2
Username: rogaha
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. docker service create --name service1 --network whoami-net -p 8000 jwilder/whoami
2. docker service scale service1=3

**Describe the results you received:**
```
rogaha@Robertos-MacBook-Pro:~$ docker service inspect service1                                                                                                              1:37:42
[
    {
        "ID": "cojb16cncgj76z0sslxt015bc",
        "Version": {
            "Index": 112
        },
        "CreatedAt": "2016-08-03T15:15:33.784675129Z",
        "UpdatedAt": "2016-08-04T08:10:52.87831274Z",
        "Spec": {
            "Name": "service1",
            "TaskTemplate": {
                "ContainerSpec": {
                    "Image": "jwilder/whoami"
                },
                "Resources": {
                    "Limits": {},
                    "Reservations": {}
                },
                "RestartPolicy": {
                    "Condition": "any",
                    "MaxAttempts": 0
                },
                "Placement": {}
            },
            "Mode": {
                "Replicated": {
                    "Replicas": 3
                }
            },
            "UpdateConfig": {
                "Parallelism": 1,
                "FailureAction": "pause"
            },
            "Networks": [
                {
                    "Target": "cke02aohtbpspx5so5gc6a76x"
                }
            ],
            "EndpointSpec": {
                "Mode": "vip",
                "Ports": [
                    {
                        "Protocol": "tcp",
                        "TargetPort": 8000
                    }
                ]
            }
        },
        "Endpoint": {
            "Spec": {
                "Mode": "vip",
                "Ports": [
                    {
                        "Protocol": "tcp",
                        "TargetPort": 8000
                    }
                ]
            },
            "Ports": [
                {
                    "Protocol": "tcp",
                    "TargetPort": 8000,
                    "PublishedPort": 30000
                }
            ],
            "VirtualIPs": [
                {
                    "NetworkID": "5wgpghefhq0indqu50kbzee3v",
                    "Addr": "10.255.0.2/16"
                },
                {
                    "NetworkID": "cke02aohtbpspx5so5gc6a76x",
                    "Addr": "10.0.0.2/24"
                }
            ]
        },
        "UpdateStatus": {
            "StartedAt": "0001-01-01T00:00:00Z",
            "CompletedAt": "0001-01-01T00:00:00Z"
        }
    }
]

rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:18:26
I'm a2531712ae05
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:19:56
I'm a2531712ae05
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:20:09
I'm a2531712ae05
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:20:09
I'm a2531712ae05
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:20:10
I'm a2531712ae05
rogaha@Robertos-MacBook-Pro:~$

rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.102:30000/                                                                                                            1:20:10
I'm b0d6d239f694
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.102:30000/                                                                                                            1:20:45
I'm b0d6d239f694
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.102:30000/                                                                                                            1:20:46
I'm b0d6d239f694
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.102:30000/                                                                                                            1:20:47
I'm b0d6d239f694
rogaha@Robertos-MacBook-Pro:~$

rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.105:30000/                                                                                                            1:20:48
I'm af339bc5e1bf
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.105:30000/                                                                                                            1:21:00
I'm af339bc5e1bf
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.105:30000/                                                                                                            1:21:01
I'm af339bc5e1bf
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.105:30000/                                                                                                            1:21:01
I'm af339bc5e1bf
rogaha@Robertos-MacBook-Pro:~$
```
**Describe the results you expected:**
```
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:18:26
I'm a2531712ae05
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:19:56
I'm b0d6d239f694
rogaha@Robertos-MacBook-Pro:~$ curl http://192.168.99.100:30000/                                                                                                            1:20:09
I'm af339bc5e1bf
```
**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by rogaha** [comment]
It might be related to networking as well, see below the details about the `whoami-net` overlay network:
```
rogaha@Robertos-MacBook-Pro:~$ docker network ls                                                                                                                            2:47:32
NETWORK ID          NAME                DRIVER              SCOPE
0ea70f2651d7        bridge              bridge              local
ca341b9644c8        docker_gwbridge     bridge              local
34cd36e32e75        host                host                local
5wgpghefhq0i        ingress             overlay             swarm
4a514a524726        none                null                local
cke02aohtbps        whoami-net          overlay             swarm
rogaha@Robertos-MacBook-Pro:~$ docker network inspect whoami-net                                                                                                            2:54:59
[
    {
        "Name": "whoami-net",
        "Id": "cke02aohtbpspx5so5gc6a76x",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.0.0.0/24",
                    "Gateway": "10.0.0.1"
                }
            ]
        },
        "Internal": false,
        "Containers": {
            "61b9fb8d79a928641c2124fb893ba27b187236d55a513d68db152298a9e68643": {
                "Name": "service2.4.9gvvljb25qrm71c1jazvx4b7d",
                "EndpointID": "381e23fddbfb9bf05f39c9c07180030db65b9327d6c968c89df98be8ece6df1a",
                "MacAddress": "02:42:0a:00:00:0a",
                "IPv4Address": "10.0.0.10/24",
                "IPv6Address": ""
            },
            "a2531712ae053b426ab6957ab034ae476a26e7f6915d5a53705a91c654aaa66d": {
                "Name": "service1.1.c83vrhg87a0hxpfo0te4weat6",
                "EndpointID": "b61138779dd6247d743015fc559ae3c4dd544c03894bd4326f8d62dca275671b",
                "MacAddress": "02:42:0a:00:00:03",
                "IPv4Address": "10.0.0.3/24",
                "IPv6Address": ""
            },
            "c37dab81741fa972b6a9ade72367545f78ae0e454a9c784454d850a40da7f321": {
                "Name": "service3.2.8dkjkhz15ov94ktec6bcq1qa1",
                "EndpointID": "39daa00813add94d648a4bd1835644e83f685cb0efc8ae564e68b89afd143709",
                "MacAddress": "02:42:0a:00:00:10",
                "IPv4Address": "10.0.0.16/24",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "257"
        },
        "Labels": {}
    }
]
rogaha@Robertos-MacBook-Pro:~$
```

**Comment by nathanleclaire** [comment]
I saw similar issue and I think @mrjana suggested it had been fixed in libnetwork upstream

**Comment by manast** [comment]
I am having the same issue as well. I don't think is related to the network, I am running the instavote/vote app, as it is demod on docker blog (https://youtu.be/F7hoq0KwHD4?t=155), and for some reason the load balancing across nodes is not working, when you access a node it only load balances to the containers that are running in that node and not in the others. I don't know yet if this issue appears only on the services with published ports or even in intra container communication.

**Comment by xiaods** [comment]
@all do we need clarify the definition on  load balancing across nodes feature?  Does IPVS support random access the entrypoint?

**Comment by xiaods** [comment]
any update?

**Comment by rogaha** [comment]
It seems to be fixed on `1.12.2-rc1`.

**Comment by rogaha** [comment]
Closing
